### PR TITLE
chore: Bump Babylon dependency and update Docker image

### PR DESCRIPTION
**Pull request by vitsalis**
This PR:
1. Updates the Babylon dependency to the current `dev` branch. (main change)
2. Makes the `Makefile` more robust and restructures the `cmd` directory.
3. Creates a restructured and unified Docker image for all different modes. This is appropriate as currently we use a copy of the same Docker image for all different modes and we wrap the command we want to execute under a `wrapper.sh` script. Here, we change this to use a single image, that we can execute according to the mode we want.

The PR is still a draft one as we wait for a new version of Babylon and rpc-client to be released, but given the amount of changes, a first round of review would be appreciated to speed up the process. Once the Babylon version is created we can have another review round.

### CODE REVIEW

File: submitter/submitter.go [hunk 1]
```diff
@@ -31,12 +36,31 @@ type Submitter struct {
 }
 
 func New(cfg *config.SubmitterConfig, btcWallet *btcclient.Client, queryClient query.BabylonQueryClient,
-	submitterAddr sdk.AccAddress, tagIdx uint8, metrics *metrics.SubmitterMetrics) (*Submitter, error) {
+	submitterAddr sdk.AccAddress, retrySleepTime, maxRetrySleepTime time.Duration, metrics *metrics.SubmitterMetrics) (*Submitter, error) {
+
```
Comment (SebastianElvis) [inline submitter/submitter.go:40]: Can we remove this empty line?

File: cmd/vigilante/cmd/utils.go [hunk 2]
```diff
@@ -3,15 +3,27 @@
 // Use of this source code is governed by an ISC
 // license that can be found in the LICENSE file.
 
-package utils
+//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris
+// +build darwin dragonfly freebsd linux netbsd openbsd solaris
```
[the remark this answers is not in the document]
Comment (vitsalis) [reply cmd/vigilante/cmd/utils.go:7]: They are copied from the `cmd/utils/signalsigterm.go` file that this PR removes.

File: cmd/vigilante/cmd/monitor.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+const (
+	genesisFileNameFlag    = "genesis"
+	GenesisFileNameDefault = "genesis.json"
+)
+
+// GetMonitorCmd returns the CLI commands for the monitor
+func GetMonitorCmd() *cobra.Command {
+	var genesisFile string
+	var cfgFile = ""
+	// Group epoching queries under a subcommand
```
Comment (KonradStaniec) [inline cmd/vigilante/cmd/monitor.go:29]: epoching ?

File: submitter/relayer/relayer_test.go [hunk 4]
```diff
@@ -25,8 +25,8 @@ package relayer_test
 //datagen.AddRandomSeedsToFuzzer(f, 10)
```
Comment (gusin13) [inline submitter/relayer/relayer_test.go:25]: why do we have these tests commented out ?

File: types/ckpt_segment_test.go [hunk 5]
```diff
@@ -34,8 +33,10 @@ func toWrappedTx(txHex string) (*btcutil.Tx, error) {
 }
 
 func TestNewCkptSegment(t *testing.T) {
-	bbnParams := netparams.GetBabylonParams(types.BtcSimnet.String(), 48)
-	tag, version := bbnParams.Tag, bbnParams.Version
+	// The transaction hex corresponds to the tat "bbt0" (in string format)
```
Comment (KonradStaniec) [inline types/ckpt_segment_test.go:36]: tat -> tag

File: submitter/submitter.go [hunk 6]
```diff
@@ -31,12 +36,31 @@ type Submitter struct {
...
+	}
+
+	// get checkpoint tag
+	checkpointTag, err := hex.DecodeString(btccheckpointParams.Params.CheckpointTag)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode checkpoint tag: %w", err)
+	}
 
 	p := poller.New(queryClient, cfg.BufferSize)
 	r := relayer.New(btcWallet,
```
Comment (gusin13) [inline submitter/submitter.go:61]: maybe out of this PR's scope but I didn't understand why we call this a relayer in submitter?

**Review by KonradStaniec — APPROVED** [review]
lgtm!

**Review by gusin13 — APPROVED** [review]

File: submitter/relayer/relayer_test.go (hunk 4 above)
Comment (vitsalis) [reply submitter/relayer/relayer_test.go:25]: These tests are TODOs.

**Pull request merged**